rareapi/views/Profiles.py

```diff
@@ -0,0 +1,45 @@
+from django.http.response import HttpResponseServerError
+from rest_framework.response import Response
+from rest_framework.viewsets import ViewSet
+from rest_framework.decorators import action
+from django.contrib.auth.models import User
+from rareapi.models import RareUsers
+from rest_framework import status
+from django.core.exceptions import ValidationError
+from rest_framework import serializers
+
+class ProfileViewSet(ViewSet):
+    def list(self, request):
+        users = RareUsers.objects.all()
+        serializer = BasicProfileSerializer(users, many=True, context={'request':request})
+        return Response(serializer.data)
+
+
+    """action to toggle between admin and author status depending on message from client"""
+    @action(methods=['patch'], detail=True)
+    def update_role(self, request, pk=None):
+        try:
+            rare_user = RareUsers.objects.get(pk=pk)
+            user = User.objects.get(pk=rare_user.id)
```

Comment:
This is still just getting the `User` by the `RareUser` id right? Instead of using the `User.objects.get` thing to get the user object can't you just say `user = rare_user.user`?

Reply:
czeckit